api/handler/multipart_upload.go

```diff
@@ -398,31 +392,48 @@ func (h *handler) CompleteMultipartUploadHandler(w http.ResponseWriter, r *http.
 			return
 		}
 
-		if len(uploadData.TagSet) != 0 {
-			t := &layer.PutTaggingParams{
-				ObjectInfo: objInfo,
-				TagSet:     uploadData.TagSet,
-			}
-			if err = h.obj.PutObjectTagging(r.Context(), t); err != nil {
-				h.logAndSendError(w, "could not put tagging file of completed multipart upload", reqInfo, err, additional...)
+		if uploadData.ACL != nil {
+			if sessionTokenSetEACL, err = getSessionTokenSetEACL(r.Context()); err != nil {
+				h.logAndSendError(w, "couldn't get eacl token", reqInfo, err)
 				return
 			}
 		}
+	}
 
-		if uploadData.ACL != nil {
-			resInfo := &resourceInfo{
-				Bucket: objInfo.Bucket,
-				Object: objInfo.Name,
-			}
-			astObject, err := aclToAst(uploadData.ACL, resInfo)
-			if err != nil {
-				h.logAndSendError(w, "could not translate acl of completed multipart upload to ast", reqInfo, err, additional...)
-				return
-			}
-			if err = h.updateBucketACL(r, astObject, bktInfo); err != nil {
-				h.logAndSendError(w, "could not update bucket acl while completing multipart upload", reqInfo, err, additional...)
-				return
-			}
+	c := &layer.CompleteMultipartParams{
+		Info:  uploadInfo,
+		Parts: reqBody.Parts,
+	}
+	objInfo, err := h.obj.CompleteMultipartUpload(r.Context(), c)
```

Comment:
Why did we move this invocation?

Reply:
Before we used to make the following steps:
1. Get init part info
2. Complete multipart upload
3. If init part has payload, get init part, unmarshal payload, put tagging and/or `ACL` file

If a user don't have a session token, step 3 fails, but an object is created. Thus, now we the following steps: 
1. Get init part info
2. If init part has payload and the payload contains `ACL`, check presence of `EACL` token.
3. Complete multipart upload
4. Put tagging and/or `ACL` file